scripts/floods/earthengine_image.py

```diff
@@ -0,0 +1,598 @@
+flags.DEFINE_string(
+    'time_period', 'P1M',
+    'Time range of images to be aggregated into a single image in P<N><T> format, '
+    'for example "P1Y" for images merged over a year')
+flags.DEFINE_string('ee_image_collection', '',
+                    'Name of the EarthEngine image collection asset.')
+flags.DEFINE_string(
+    'ee_reducer', 'mean',
+    'EarthEngine Reducer to be applied when merging images in the collection.')
+flags.DEFINE_string('band', '', 'Band to be extracted from the ee image.')
+flags.DEFINE_integer('scale', 1000, 'Scale in meters for the output image.')
+flags.DEFINE_string(
+    'ee_mask', '', 'Dataset to be applied as a mask when extracting the image.')
+flags.DEFINE_float('band_min', None,
+                   'Minimum value for the pixels in the band.')
+flags.DEFINE_float('band_max', None,
+                   'Minimum value for the pixels in the band.')
+flags.DEFINE_float('band_eq', None, 'Value for the pixels in the band.')
+flags.DEFINE_bool(
+    'ee_band_bool', True,
+    'Filter band by thresholds into a bool image with 1 where values are met.'
+    'If False, the originals values meeting thresholds are preserved, '
+    'and image set to 0 everywhere else.')
+flags.DEFINE_string(
+    'ee_image_eval', None,
+    'Get image by evaluating the python statements with ee commands.')
+flags.DEFINE_string(
+    'ee_bounds', None,
+    'Rectangular bounds as "lat,lng:lat,lng" for filtering images.')
+flags.DEFINE_bool('debug', False, 'Enable debug messages.')
+flags.DEFINE_integer(
+    'image_count', 1,
+    'Number of images to generate, each advanced by --time_period.')
+
+_FLAGS = flags.FLAGS
+_FLAGS(sys.argv)  # Allow invocation without app.run()
+
+# Default Earth Engine datasets with asset name and bands.
+_DEFAULT_DATASETS = {
+    # Land mask from Hansen global forest cover 2015
+    'land': {
+        'ee_image': 'UMD/hansen/global_forest_change_2015',
+        'band': 'datamask',
+        'band_eq': 1,
+    },
+
+    # Water band from dynamic world
+    # https://developers.google.com/earth-engine/datasets/catalog/GOOGLE_DYNAMICWORLD_V1#bands
+    'dynamic_world': {
+        'ee_image_collection': 'GOOGLE/DYNAMICWORLD/V1',
+        # 'band': 'water',
+        # 'band_min': 0.7, # Probability of a pixel to be water
+    },
+
+    # Water from Sentinel-1 SAR GRD
+    'sar': {
+        # Get image with band VV
+        'ee_image_eval':
+            "ee.ImageCollection('COPERNICUS/S1_GRD').filter(ee.Filter.listContains('transmitterReceiverPolarisation', 'VV')).filter(ee.Filter.eq('instrumentMode', 'IW')).select('VV')",
+        # 'band': 'VV',
+        # 'band_max': -17,
+    },
+
+    # NASA FIRMS Fires
+    # https://developers.google.com/earth-engine/datasets/catalog/FIRMS
+    'fires': {
+        'ee_image_collection': 'FIRMS',
+        # 'band': 'T21',  # Brightness temperature of a fire pixel in K
+    },
+}
+
+EE_DEFAULT_CONFIG = {
```

Comment:
Comment what this is?  I'm puzzled why the flags are set to a global map?  Why not just skip flags and pass in a json file with these params?

Reply:
The commandline flags are more for a local run overriding a few settings from the config file.
Something like --vars=. 

Removed the settings flags and added an override option with single --config_params='...'.